modules/basics/src/main/java/com/opengamma/strata/basics/date/ImmutableHolidayCalendar.java

```diff
@@ -606,6 +613,49 @@ private LocalDate lastBusinessDayOfMonthOutOfRange(LocalDate date) {
     throw new IllegalArgumentException("Date is outside the accepted range (year 0000 to 10,000): " + date);
   }
 
+  //-------------------------------------------------------------------------
+  @Override
+  public int daysBetween(LocalDate startInclusive, LocalDate endExclusive) {
+    ArgChecker.inOrderOrEqual(startInclusive, endExclusive, "startInclusive", "endExclusive");
+    try {
+      // find data for start and end month
+      int startIndex = (startInclusive.getYear() - startYear) * 12 + startInclusive.getMonthValue() - 1;
+      int endIndex = (endExclusive.getYear() - startYear) * 12 + endExclusive.getMonthValue() - 1;
+      
+      // count of first month = ones after day of month inclusive
+      // e.g 4th day of month - want holidays from index 3 inclusive
+      int start = Integer.bitCount(lookup[startIndex] & (BIT_MASK_ALL_ONES << (startInclusive.getDayOfMonth() - 1)));
+      // count of last month = ones before day of month exclusive == total for month - ones after end inclusive
+      int missingEnd = Integer.bitCount(lookup[endIndex] & (BIT_MASK_ALL_ONES << endExclusive.getDayOfMonth() - 1));
+      if (startIndex == endIndex) {
+        // same month - return holidays up to end exclusive 
+        return start - missingEnd;
+      }
+      
+      int end = Integer.bitCount(lookup[endIndex]) - missingEnd;
+      // otherwise add start and end month counts, and sum months between
+      long count = start + end;
```

Comment:
My maths suggests `int` is sufficient here, as the maximum days in 10,000 years is within capacity